### PR TITLE
Standardized Fill Values and Data Types for Connectivity Variables

### LINKED ISSUE #239: Add Standardized Data Types and Fill Values for UGRID connectivity variables
Use a standardized data type and fill values for connectivity variables. Continutation of #189

### PR DESCRIPTION
**Pull request by philipc2**
Closes #239, #189

### Overview 
* Introduces the `constants.py` module, which currently only stores the standardized integer data type `INT_DTYPE` and fill value `INT_FILL_VALUE` which is derived from the integer data type
* As discussed in #189, the standard data type for integer indexing variables will be set to `uintp`, with a corresponding fill value of `max(uintp`)
* `Mesh2_face_nodes` is set to always be of type `uintp` (either `uint32` or `uint64` depending on system) and use the fill value as described above
* Changes to I/O modules `_scrip.py`, `_ugrid.py` and `_exodus.py` were made to correctly represent `Mesh2_face_nodes`
* Introduction of the `replace_fill_values()` helper function, which can be used on any grid connectivity variable to replace all instances of a face value with a new one, with the addition of converting data types 

### Expected Use
* With the fill value now being standardized, they can be either accessed through the `_FillValue` attribute under `Mesh2_face_nodes` or through `uxarray.constants.INT_FILL_VALUE`
* Additional connectivity variables, such as `Mesh2_edge_nodes`, should adhere to the standards set in `uxarray.constants`

### CODE REVIEW

File: uxarray/constants.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import numpy as np
+
+# is 32-bit safe or should we use 64-bit?
+INT_DTYPE = np.uint32
+FILL_VALUE = np.iinfo(INT_DTYPE).max
```
Comment (rljacob) [inline uxarray/constants.py:5]: Should this be INTFILL_VALUE ?   What are doing for data on the grid that is in float and comes with a _FillValue?  Do we still want those to be Nan?

Comment (rljacob) [reply uxarray/constants.py:5]: In https://github.com/UXARRAY/uxarray/issues/189#issuecomment-1428509976 you thought we'd need FILL_VALUE32 and FILL_VALUE64.

Comment (philipc2) [reply uxarray/constants.py:5]: Good point, we should specify that this fill value is for integer arrays, I like the name `INT_FILL_VALUE`

The `replace_fill_values()` method handles converting data types and pre-defined fill values to the standardized versions, with correct edge case handling.

I think `FLOAT_FILL_VALUE` should be set to `NaN`, but this PR is only standardizing the connectivity variables, which should never be floats. Once we get these changes merged we can standardize the float variables to use a consistent fill value.

Comment (philipc2) [reply uxarray/constants.py:5]: > In [#189 (comment)](https://github.com/UXARRAY/uxarray/issues/189#issuecomment-1428509976) you thought we'd need FILL_VALUE32 and FILL_VALUE64.

After giving it some more thought, for connectivity variables I don't see a situation where we'd need more than 32-bit precision. However, we could use [`np.uintp`](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.intp:~:text=%27p%27-,numpy.uintp,%23,-Alias%20for%20the) which would dynamically change to either 32 or 64 bit depending on the system.

**Review by erogluorhan — COMMENTED** [review]
Although I like this overall, I have a few in-line comments

File: uxarray/helpers.py [hunk 2]
```diff
@@ -556,3 +557,45 @@ def _normalize_in_place(node):
         raise RuntimeError("Input array should have a length of 3: [x, y, z]")
 
     return list(np.array(node) / np.linalg.norm(np.array(node), ord=2))
+
+
+def replace_fill_values(grid_var, original_fill, new_fill, new_dtype=None):
```
[the remark this answers is not in the document]
Comment (philipc2) [reply uxarray/helpers.py]: 1. Changing it to an internal helper is safer, change made is most recent commit 
2. Will work on this in the redesign branch, current location is fine for now

**Review by erogluorhan — APPROVED** [review]
Approving this per its scope, but please be sure to create a corresponding issue to keep track of the broader requests made in this convo (e.g. floating point, moving from helpers.py to a better place, etc). Thanks for this!

**Pull request merged**